### ISSUE TITLE
Proposal: SA1027 should use tabs for indentation when useTabs is true

### ISSUE THREAD

**Issue by sharwell** [DotNetAnalyzers/StyleCopAnalyzers#2037]
~~**Blocked on the acceptance of #2039, which adds the option described here.**~~

Currently SA1027 checks for tabs and converts them to spaces. When `useTabs` is true, I propose automatically having SA1027 switch to checking for spaces used for indentation, reporting a diagnostic, and automatically converting them to tabs according to the `tabSize` property.

**Comment by vweijsters** [comment]
:+1: We may need to think about rounding policy. If the indent is 7 spaces and the tab size is 4, what would the code fix do? Replace with 1 tab or 2 tabs or 1 tab + 3 spaces?

**Comment by sharwell** [comment]
@vweijsters We do the same thing the IDE does. For the example you gave, it would be 1 tab + 3 spaces.

**Comment by sharwell** [comment]
@vweijsters Note that SA1027 would not use the `indentationSize` property. Any line would be allow to start with any number of tabs, followed by up to (`tabSize` - 1) spaces, regardless of the value of `indentationSize`.

**Comment by sharwell** [comment]
This proposal is no longer blocked.

**Comment by AArnott** [comment]
My only concern with this proposal is that SA1027's name suggests it's a "use spaces" policy. If I get warnings from SA1027 and the name says "use spaces" but the message says "use tabs", won't that be more confusing than if we just had SX1027 with a new name?

**Comment by sharwell** [comment]
@AArnott If accepted, the messaging would change. The change I'm proposing is in https://github.com/sharwell/StyleCopAnalyzers/commit/a42e61977477ddd56962c65ed7c96031c43525ef.

**Comment by AArnott** [comment]
That sounds reasonable. :+1:

**Comment by Noryoko** [comment]
:+1:

**Comment by Trass3r** [comment]
Does that also support a hybrid way of using tabs for indentation and spaces for alignment (in statement spanning multiple lines)?

**Comment by AArnott** [comment]
I hope so. I think there was another issue about this... somewhere.
